garage/spaces/box.py

```diff
@@ -26,10 +33,29 @@ def __init__(self, low, high, shape=None):
             self.low = low + np.zeros(shape)
             self.high = high + np.zeros(shape)
 
+        if dtype is None:
```

Comment:
nah, there should not be two defaults based on low/high. that's mighty confusing. it also encourages people to code assuming that the framework will magically resolve their types by inspecting low/high. that creates a really difficult-to-support API.

there should be one default (np.float32), plus a warning when it seems like this isn't what the user wanted (e.g. low/high = [0, 255], warn them that we're using np.float32 anyway, but that they probably wanted np.uint8).

Reply:
so we are having np.float32 as default but if low/high=[0, 255], we use np.uint8 and warn user?